src-gui/src/main/java/org/openkilda/service/StatsService.java

```diff
@@ -515,6 +523,25 @@ private VictoriaData buildVictoriaData(VictoriaDbRes dbData, String metricName)
                 .build();
     }
 
+    private VictoriaData buildVictoriaData(VictoriaDbRes dbData, String metricName) {
+        MetricValues metricValues = null;
+        if (dbData.getData() != null && isNotEmpty(dbData.getData().getResult())) {
+            metricValues = dbData.getData().getResult().get(0);
+        }
+        return buildVictoriaData(dbData, metricName, metricValues);
+    }
+
+    private List<VictoriaData> buildVictoriaDataList(VictoriaDbRes dbData, String metricName) {
+        List<VictoriaData> result = new ArrayList<>();
+
+        if (dbData.getData() != null && isNotEmpty(dbData.getData().getResult())) {
+            dbData.getData().getResult().stream()
+                    .map(metricValues -> buildVictoriaData(dbData, metricName, metricValues))
+                    .forEach(result::add);
+        }
+        return result;
+    }
```

Comment:
Much better now!
Still you can do this:

```java
  private List<VictoriaData> buildVictoriaDataList(VictoriaDbRes dbData, String metricName) {
      if (dbData.getData() == null) {
          return Collections.emptyList();
      }
      return dbData.getData().getResult().stream()
              .map(metricValues -> buildVictoriaData(dbData, metricName, metricValues))
              .collect(Collectors.toList());
  }
```
So, you don't have to assign variables. And the isNotEmpty(dbData.getData().getResult()) is not necessary because if it's empty, then stream is gonna be iterated. I think it looks nicer.

Reply:
true. fixed!